app/Http/Controllers/App/PostController.php

```diff
@@ -132,6 +141,29 @@ public function update(Request $request, Post $post)
         $post->content = $data['content'];
         $post->banner_id = $data['banner_id'];
         $post->challenge_id = $data['challenge_id'];
+        $post->video = $data['video'] ?? '';
+
+        if(isset($data['removeImage']) && $data['removeImage'] && $post->image){
+            Storage::disk('upload')->delete($post->image);
+            $post->image = '';
+        }
+
+        if (!empty($data['image'])) {
```

Comment:
Wenn ein neues Bild hochgeladen wird, sollte auch das alte gelöscht werden.

Reply:
wenn man in der GUI ein neues Bild auswählen möchte muss man vorher auf "Bild entfernen" klicken. Dadurch wird das Flag zum Löschen des alten Bildes in den Form-Daten gesetzt.
Ist nicht offensichtlich, sollte aber funktionieren.

Reply:
> wenn man in der GUI ein neues Bild auswählen möchte muss man vorher auf "Bild entfernen" klicken.

Das ist noch nicht der Fall glaube ich, ich kann ein neues bild wählen, ohne das alte zu löschen:

![2021-08-24-220600](https://user-images.githubusercontent.com/3864879/130682586-bc23a339-2260-4641-ab79-f8251d72679e.png)
![2021-08-24-220608](https://user-images.githubusercontent.com/3864879/130682590-b39f1e5d-fc35-41fa-af16-deccea19922b.png)

Reply:
Oh da hat eine Bedingung gefehlt. Jetzt sollte es passen.